neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -125,8 +125,13 @@ public override int GetHashCode()
 
         public UInt160[] GetScriptHashesForVerifying(Snapshot snapshot)
         {
-            var hashes = new HashSet<UInt160> { Sender };
-            hashes.UnionWith(Cosigners.Select(p => p.Account));
+            if (Cosigners.Length == 0 || (Cosigners.Length == 1 && Cosigners[0].Account == Sender))
+            {
+                return new UInt160[] { Sender };
+            }
+
+            var hashes = new HashSet<UInt160>(Cosigners.Select(p => p.Account));
+            hashes.UnionWith(new HashSet<UInt160> { Sender });
```

Comment:
I think there is no difference in this place.

Reply:
> I think there is no difference in this place.

Yes, length of Cosigners cannot be too long.

Reply:
I prefer to maintain the if where only `Sender` is returned

Reply:
> I prefer to maintain the if where only `Sender` is returned

Yes I think so. Another question, is it possible for such occasions to exist, that Cosigners be so long (i.e. 1M) that the computation complexity of func UnionWith must be taken into account? If not maybe this part of change is not necessary.